shared-bindings/supervisor/__init__.c

```diff
@@ -109,6 +110,29 @@ STATIC mp_obj_t supervisor_set_next_stack_limit(mp_obj_t size_obj) {
 }
 MP_DEFINE_CONST_FUN_OBJ_1(supervisor_set_next_stack_limit_obj, supervisor_set_next_stack_limit);
 
+STATIC mp_obj_t supervisor_set_next_code_file(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_filename, MP_ARG_REQUIRED | MP_ARG_OBJ, {.u_rom_obj = mp_const_none} },
+        { MP_QSTR_reload_on_success, MP_ARG_KW_ONLY | MP_ARG_BOOL, {.u_bool = false} },
```

Comment:
What is this for? Please add a comment with the python stub. (Lines that start with `//|`)

Reply:
My plan was to add documentation later when we have a consensus on how things should work and it’s less of a moving target. Does what I wrote at the bottom of https://github.com/adafruit/circuitpython/issues/1084#issuecomment-698851943 explain it sufficiently for now? Otherwise I’ll add the documentation now.

Reply:
I actually prefer the opposite. Doing the docs first lets me brainstorm the API with examples that don't work and function docs that force me to explain how things should work.

I think adding docs now will help clarify what exactly we're doing for the game case.

Reply:
I'm with @tannewt on the docs first or even _tests_ first but that's a bit harder of a pill for some to swallow. Having an explanation of how the code should work before you write it makes you consider how you want it to work in more detail than you might if you jumped in. Additionally it is helpful when collaborating because it gives you a concrete example to discuss and make changes _before_ you've done the work

Reply:
All fair points, but I’m not here to argue about that. For the purpose of this discussion, the point is moot because the docs are there now. (I can look into adding tests.)